# JHU EN.600.649: Computational Genomics: Applied Comparative Genomics
## Project Presentations

Presentations will be a total of 15 minutes: 12 minutes for the presentation, followed by 3 minutes for questions. We will strictly keep to the schedule to ensure that all groups can present in class. 

## Schedule of Presentations

Day     | Time      | Team Name               | Students                                    | Title 
--------|-----------|-------------------------|--------------------------------------------|---------------------------------------------------------------------
Th 3/27 | 2:30-2:45 | Kayarash                | Kayarash Karimian                          | Comparing the accuracy of PacBio aligners as a function of error rate
   .    |     .     |            .            |                       .                    |                              .
Tu 5/2  | 1:30-1:45 | STaY                    | Sam Kovaka, Taher Mun, Yunfan Fan          | Base call free nanopore read alignment: Aligning nanopores against the reference
Tu 5/2  | 1:45-2:00 | Rachel                  |	Rachel Sherman                             | Incorporating SVs into Phased Genome Assemblies
Tu 5/2  | 2:00-2:15 | Bang for Your Buck      | Isac Lee, Suraj Kanaan, Andrew Fraser      | Efficient Selection for Marks in Epigenomic Analysis
Tu 5/2  | 2:15-2:30 | Thank God it’s Genomics	| Shubhi Bartaria, Saranya Akumalla          | Test and apply LACHESIS on long range information from Hi-C
Tu 5/2  | 2:30-2:45 | BS 649                  | Bayan Al Muhander, Shubha Tirumale         | Benchmarking of non-coding mutation analysis schemes on diseased genomes
   .    |     .     |            .            |                       .                    |                              .
Th 5/4  | 1:30-1:45 | Charlotte               | Charlotte Darby                            | The role of mutations in computer binaries
Th 5/4  | 1:45-2:00 | Spradling Lab          	| Liang-Yu Pang                              | Identify deletion sites in polyploid Drosophila follicle cell genome.
Th 5/4  | 2:00-2:15 | DeepWorker              | Guangyu Yang                               | Experiments and Extension on DeepVariant
Th 5/4  | 2:15-2:30 | GenomeScope 2           | Ravi Gaddipati, Rhyker Ranallo-Benavidez   | EGSP: Extending GenomeScope Ploidy
Th 5/4  | 2:30-2:45 | Rock & “Role”          	| Gherman Uritskiy, Peter DeFord, Xiuqi Chen | Investigating symbiotic roles for nutrient acquisition in a metatranscriptome analysis in rock colonies from the Atacama Desert
    
<br>
<br>

## Recommended outline for your talk (1 minute per slide):

1. Title Slide: Who are you, title, date
2. Intro 1: Whats the big idea???
3. Intro 2: More specifically, what are you trying to learn?
4. Methods 1: What did you try?
5. Methods 2: What is the key idea?
6. Data 1: What data are you looking at?
7. Data 2: Anything notable about the data?
8. Results 1: What did you see!
9. Results 2: Does it work?
10. Results 3: How does it compare to other methods/data/ideas?
11. Discussion 1: What did you learn from this study?
12. Discussion 2: What does this mean for the future?
13. Acknowledgements: Who helped you along the way?

I strongly *discourage* you from trying to give a live demo as they are too unpredictable for a short talk. If you have running software you want to show, use a "cooking show" approach, where you have screen shots of the important steps.    
    
